src/bin/sol-fbp-runner/main.c

```diff
@@ -149,12 +181,208 @@ parse_args(int argc, char *argv[])
+
+                        SOL_JSON_SCANNER_OBJECT_LOOP_NEST (&entries, &token, &key, &value, reason) {
+                            if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "name")) {
+                                sol_json_token_remove_quotes(&value);
+                                name = sol_json_token_to_slice(&value);
+                                if (!name.len) {
+                                    SOL_WRN("Couldn't get entry name");
+                                    goto error;
+                                }
+                            } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "offset")) {
+                                if (sol_json_token_get_uint32(&value, &offset) < 0) {
+                                    SOL_WRN("Couldn't get entry offset");
+                                    goto error;
+                                }
+                            } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "size")) {
+                                if (sol_json_token_get_uint32(&value, &size) < 0) {
+                                    SOL_WRN("Couldn't get entry size");
+                                    goto error;
+                                }
+                            } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "bit_offset")) {
+                                if (sol_json_token_get_uint32(&value, &bit_offset) < 0) {
+                                    SOL_WRN("Couldn't get entry size");
+                                    goto error;
+                                }
+                                if (bit_offset > 7) {
+                                    SOL_WRN("Entry bit offset cannot be greater than 7");
+                                    goto error;
+                                }
+                            } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&key, "bit_size")) {
+                                if (sol_json_token_get_uint32(&value, &bit_size) < 0) {
+                                    SOL_WRN("Couldn't get entry bit size");
+                                    goto error;
+                                }
+                            }
+                        }
+                        if (reason != SOL_JSON_LOOP_REASON_OK)
+                            goto error;
+
+                        if (bit_size > (size * 8)) {
+                            SOL_WRN("Invalid bit size for entry. Must not be greater"
+                                "than size * 8 [%d]", size * 8);
+                            goto error;
+                        }
+                        if (!size) {
+                            SOL_WRN("Invalid size for entry");
+                            goto error;
+                        }
+
+                        ptr_table_entry = sol_vector_append(&entries_vector);
+                        SOL_NULL_CHECK_GOTO(ptr_table_entry, error);
+
+                        memmap_entry = calloc(sizeof(struct sol_memmap_entry), 1);
```

Comment:
change the order of parameters, to be consistent with other paths in your code.

and move `if (!name.len)` befre this `calloc()` and `sol_vector_append()` (fail as soon as possible).